File: Pow(x,n).py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        if n<0:
            x = 1/x
            n = -n
            
        res = 1
        while n:
            if n & 1:
                res *= x
                n -= 1
            else:
                x *= x
                n >>= 1
        return res
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        # if n <0:
        #     n = -n
        #     x = 1/x
        # pow = 1
        # while n:
        #     if n & 1:
        #         pow *= x
        #     x *= x
        #     n >>= 1
        # return pow

File: pow(x,n).kt
class Solution {
    fun myPow(xval: Double, nval: Int): Double {
        var n: Long = nval.toLong()
        var x = xval
        if(n < 0){
            x = 1 / x
            n = -n
        }
        var res: Double = 1.0
        while(n > 0){
            when{
                (n and 1) == 1L -> {
                    res *= x
                    n --
                }
                else -> {
                    x *= x
                    n = n shr 1
                }
            }
            // println("n= " + n + " res= " + res+ " x= " + x)
        }
        return res
    }
}
